src/commands/deploy/failureMoreInfoSurvey.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { env, ExtensionContext, Uri, window } from 'vscode';
+import { localize } from "../../localize";
+
+const SURVEY_URL = 'https://aka.ms/AppServiceExtDeploymentFeedback';
+
+const NEVER_AGAIN_KEY = 'failureMoreInfoSurvey/neverAgain';
```

Comment:
The typedef stuff is pretty trivial. For example here, all you need to do here is:
```
const NEVER_AGAIN_KEY: string = 'failureMoreInfoSurvey/neverAgain';
```

If you merge with master, the lint rule is enabled now. https://github.com/microsoft/vscode-azureappservice/pull/1788

The only weird type is your "take"/"never" buttons with "run". Normally we would do it like [this](https://github.com/microsoft/vscode-azureappservice/blob/b4cec5cc748e23833e191645ea295e3cc0d7f28a/src/commands/deploy/promptScmDoBuildDeploy.ts#L22) or [this](https://github.com/microsoft/vscode-azureappservice/blob/b4cec5cc748e23833e191645ea295e3cc0d7f28a/src/commands/deploy/promptToSaveDeployDefaults.ts#L20), where the button (of type `MessageItem`) is separate from what happens after you click it